### PR TITLE
Add extra logs, fix best filter bug

### LINKED ISSUE #1617: Running appServer/run causes computer to crash or hang on startup
**Describe the bug**

When running off of 0d52199699e42c4e0e3db610f3c84fed8e932546 I have this behavior where my node either

1. Freezes my computer until the process is killed by the OS (Scala 2.13)
2. Does some work, but doesn't actually start syncing the node (Scala 2.12)

### Freezing computer

This seems to happen on Scala 2.13, I have my suspicions that this is related to #1521 

### Hangs on startup

I'm not sure what this is, it happens on Scala 2.12, we need to add more logs to indicate when individual submodules are initialized. Perhaps we can use `StartStop` interface to do this for modules?

Here is logs from the startup, the last line is where I seem to hang

>2020-06-26T19:17:18 INFO [WALLET] Using pre-existing wallet
2020-06-26T19:17:20 INFO [P2P] Sending getdata=GetDataMessage(inventoryCount=1, inventories=Vector(Inventory(MsgBlock,DoubleSha256Digest(b639ca945ef5833f8254988969906555654d27da080015000724210...) to peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-06-26T19:17:20 DEBUG [P2P] Sending msg=getdata to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-06-26T19:17:20 ERROR [P2P] Cannot send a message to our peer when we are not connected! payload=GetDataMessage(inventoryCount=1, inventories=Vector(Inventory(MsgBlock,DoubleSha256Digest(b639ca945ef5833f8254988969906555654d27da080015000724210...) peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-06-26T19:17:20 INFO [P2P] Starting node
2020-06-26T19:17:21 INFO [P2P] Attempting to connect to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-06-26T19:17:22 DEBUG [P2P] Tcp connection to: neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-06-26T19:17:22 DEBUG [P2P] Local: /192.168.1.187:50638
2020-06-26T19:17:22 INFO [P2P] Connection established with peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-06-26T19:17:22 DEBUG [P2P] Sending msg=version to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-06-26T19:17:22 DEBUG [P2P] Received 151 TCP bytes
2020-06-26T19:17:22 DEBUG [P2P] Parsed 2 message(s) from bytes: NetworkMessage(NetworkHeader(TestNet3,version,UInt32Impl(103),ByteVector(4 bytes, 0x83fca122)), VersionMessage(ProtocolVersion70015, ServiceIdentifier(network=true, compactFilters=true, getUtxo=false, bloom=false, witness=true, xthin=false, networkLimited=true), epoch=1593199042, receiverServices=/142.147.59.88, receiverAddress=/142.147.59.88, receiverPort=50638), userAgent=/Satoshi:0.18.99/, startHeight=1773997, relay=true)), NetworkMessage(NetworkHeader(TestNet3,verack,UInt32Impl(0),ByteVector(4 bytes, 0x5df6e0e2)), VerAckMessage)
2020-06-26T19:17:22 DEBUG [P2P] Unaligned bytes after this: 0
2020-06-26T19:17:22 DEBUG [P2P] Received message=version from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Initializing 
2020-06-26T19:17:22 DEBUG [P2P] Sending msg=verack to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-06-26T19:17:22 DEBUG [P2P] Received message=verack from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Initializing 
2020-06-26T19:17:22 INFO [P2P] Our peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) has been initialized
2020-06-26T19:17:22 INFO [P2P] Our node has been full started. It took=1638ms
2020-06-26T19:17:22 DEBUG [chain-verification] Querying for best block hash
2020-06-26T19:17:22 DEBUG [chain-verification] Querying for best block hash
2020-06-26T19:17:22 DEBUG [chain-verification] Looking for header by hash=DoubleSha256BDigestBE(000000000eb141f16b227064410339bd623efe96e0a17be9301ac35984456f4c)
2020-06-26T19:17:22 DEBUG [chain-verification] getHeader result: height=14442, hash=DoubleSha256BDigestBE(000000000eb141f16b227064410339bd623efe96e0a17be9301ac35984456f4c)
2020-06-26T19:17:22 DEBUG [chain-verification] Querying for filter count
2020-06-26T19:17:22 DEBUG [P2P] Received message=sendheaders from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal 
2020-06-26T19:17:22 DEBUG [P2P] Sending msg=sendheaders to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-06-26T19:17:22 DEBUG [P2P] Received 119 TCP bytes
2020-06-26T19:17:22 DEBUG [P2P] Parsed 3 message(s) from bytes: NetworkMessage(NetworkHeader(TestNet3,ping,UInt32Impl(8),ByteVector(4 bytes, 0x927529e6)), PingMessageImpl(UInt64Impl(6855306871668927420))), NetworkMessage(NetworkHeader(TestNet3,addr,UInt32Impl(31),ByteVector(4 bytes, 0xa02fc13b)), AddrMessageImpl(CompactSizeUInt(1),List(NetworkIpAddressImpl(UInt32Impl(1593199042),ServiceIdentifier(network=true, compactFilters=true, getUtxo=false, bloom=false, witness=true, xthin=false, networkLimited=true),/34.210.154.40,18333)))), NetworkMessage(NetworkHeader(TestNet3,feefilter,UInt32Impl(8),ByteVector(4 bytes, 0xe80fd19f)), FeeFilterMessageImpl(SatoshisPerKiloByte(-1728537831980138496 sats)))
2020-06-26T19:17:22 DEBUG [P2P] Unaligned bytes after this: 0
2020-06-26T19:17:22 DEBUG [P2P] Received message=ping from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal 
2020-06-26T19:17:22 DEBUG [P2P] Sending msg=pong to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-06-26T19:17:22 DEBUG [P2P] Received message=addr from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal 
2020-06-26T19:17:22 DEBUG [P2P] Received message=feefilter from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal

### PR DESCRIPTION
**Pull request by benthecarman**
Fixes #1617

### CODE REVIEW

**Comment by Christewart** [comment]
CI fails: https://travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/704417359#L709

File: chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala [hunk 1]
```diff
@@ -119,10 +119,48 @@ case class CompactFilterDAO()(implicit
   }
 
   /** Gets the heaviest filter from the database */
+  def getBestFilter1: Future[Option[CompactFilterDb]] = {
```
Comment (Christewart) [inline chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala]: better name? Why `getBestFilter1`?

File: chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala [hunk 2]
```diff
@@ -119,10 +119,48 @@ case class CompactFilterDAO()(implicit
...
+        if (blockHeaderDbs.isEmpty) {
+          Future.successful(None)
+        } else {
+          val blockHeaderDb = blockHeaderDbs.maxBy(_.chainWork)
+          findByBlockHash(blockHeaderDb.hashBE)
+        }
+    } yield filterHeaderOpt
+  }
+
+  implicit private val bigIntMapper: BaseColumnType[BigInt] =
```
Comment (Christewart) [inline chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala]: This should probably live some where in `dbCommons` no? Is this specific to `CompactFilterDAO`?

File: chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala (hunk 1 above)
Comment (benthecarman) [reply chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala]: meant to remove this

File: chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala (hunk 2 above)
Comment (benthecarman) [reply chain/src/main/scala/org/bitcoins/chain/models/CompactFilterDAO.scala]: These are needed for the `BlockHeaderDAO`, for the chain work we have different ways to serialize the `BitInt` for the db for postgres vs sqlite. We need to set the mapper based off of the config, but we can't get access to the config in `dbCommons` so we need to set it here. These were there beforehand I just had moved them on accident.

**Pull request merged**